src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -0,0 +1,750 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "DxFontRenderData.h"
+
+static constexpr float POINTS_PER_INCH = 72.0f;
+static constexpr std::wstring_view FALLBACK_FONT_FACES[] = { L"Consolas", L"Lucida Console", L"Courier New" };
+static constexpr std::wstring_view FALLBACK_LOCALE = L"en-us";
+
+using namespace Microsoft::Console::Render;
+
+DxFontRenderData::DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwriteFactory) noexcept :
+    _dwriteFactory(dwriteFactory),
+    _glyphCell{},
+    _lineMetrics({}),
+    _boxDrawingEffect{}
+{
+}
+
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextAnalyzer1> DxFontRenderData::Analyzer() noexcept
+{
+    return _dwriteTextAnalyzer;
+}
+
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFallback> DxFontRenderData::SystemFontFallback()
+{
+    if (!_systemFontFallback)
+    {
+        ::Microsoft::WRL::ComPtr<IDWriteFactory2> factory2;
+        THROW_IF_FAILED(_dwriteFactory.As(&factory2));
+        factory2->GetSystemFontFallback(&_systemFontFallback);
+    }
+
+    return _systemFontFallback;
+}
+
+[[nodiscard]] std::wstring DxFontRenderData::UserLocaleName()

```

Comment:
Inspired by `SystemFontFallback`, I think the `UserLocaleName` is also a system-level "constant" which we can cache and use later.

Reply:
My only worry here is that if you go into the Settings panel and change your locale, did it update live before you were caching this? Or was a logout/login or reboot required anyway?

Reply:
It should update live. You're right. I just added the `clear` to make sure it gets updated.